svc.py

```diff
@@ -0,0 +1,56 @@
+from sklearn import svm
+import sys
+import numpy as np
+
+feature_file = sys.argv[1]
+
+features_files = np.load(feature_file)
+
+features = features_files['features']
+filenames = features_files['filenames']
+labels = features_files['labels']
+
+dim = labels.shape[0]
+#two labels only
+# 0 is banana, 1 is non-banana
+twoLabels = labels
+for i in xrange(0, dim):
+    if labels[i] == 1 or labels[i] == 2:
+        twoLabels[i] = 0
+    elif labels[i] == 3:
+        twoLabels[i] = 1
+
+featuresAndLabels = np.concatenate((features, twoLabels.reshape(dim, 1)), axis=1)
+allInfo = np.concatenate((featuresAndLabels, filenames.reshape(dim, 1)), axis=1)
+
+# shuffle the order of the records randomly
+# print allInfo
+# print ""
+np.random.shuffle(allInfo)
+# print allInfo
+
+#training data
+X = allInfo[0:dim-100, 0:4095]
```

Comment:
You're losing a dimension here, aren't you? See:
```
>>> X = allInfo[0:dim-100, 0:4095]
>>> X.shape
(652, 4095)
>>> allInfo.shape
(752, 4098)
```
```X.shape``` should be ```4096``` wide. The slice doesn't include the last index. Imagine our vector was length 4, and the last 2 items were undesired:
```
>>> a = [1, 2, 3, 4]
>>> a[0:1]
[1]
>>> a[0:2]
[1, 2]
```